runtime/core/exec_aten/util/tensor_util.cpp

```diff
@@ -0,0 +1,54 @@
+#include "tensor_util.h"
+
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/runtime/core/exec_aten/util/tensor_util.h>
+
+#include <executorch/runtime/platform/assert.h>
+
+namespace executorch::runtime {
+/**
+ * Shared implementation for tensor_util.h, may only contain code that
+ * works whether or not ATen mode is active.
+ */
+std::array<char, kTensorShapeStringSizeLimit> tensor_shape_to_c_string(
+    executorch::runtime::Span<const executorch::aten::SizesType> shape) {
+  std::array<char, kTensorShapeStringSizeLimit> out;
+  char* p = out.data();
+  if ET_UNLIKELY (shape.size() > kTensorDimensionLimit) {
+    static constexpr char kLimitExceededError[] =
+        "(ERR: tensor ndim exceeds limit, can't happen)";
+    static_assert(sizeof(kLimitExceededError) <= kTensorShapeStringSizeLimit);
+    std::memcpy(p, kLimitExceededError, sizeof(kLimitExceededError));
+    return out;
+  }
+  *p++ = '(';
+  for (const auto elem : shape) {
+    if (elem < 0 || elem > internal::kMaximumPrintableTensorShapeElement) {
+      static_assert(
+          internal::kMaximumPrintableTensorShapeElement > 99999,
```

Comment:
Took me a minute to understand where `>99999` came from: please add a comment pointing out that it's the same number of digits as "ERR, " with the NUL